homeassistant/helpers/check_config.py

```diff
@@ -170,9 +170,16 @@ async def _get_integration(
         core_config = {}
 
     # Merge packages
-    await merge_packages_config(
-        hass, config, core_config.get(CONF_PACKAGES, {}), _pack_error
-    )
+    try:
+        await merge_packages_config(
+            hass, config, core_config.get(CONF_PACKAGES, {}), _pack_error
+        )
+    except vol.Invalid as err:
+        result.add_error(
```

Comment:
Looks like this case is untested.

Reply:
Hardly without changing the flow of this method. The problem here is that `merge_packages_config` raises `Invalid` only if `PACKAGES_CONFIG_SCHEMA(packages)` inside that function fails. This will not happen in `async_check_ha_config_file` because a couple lines above, `core_config = CORE_CONFIG_SCHEMA(core_config)` already transitively performs the same checks that would lead to a failure during merge. (And knowing about the double checks is merely an implementation details that one cannot necessarily derive from the function signature and docs, so in any case I would leave the exception handler in place).


We could merge the two exception handlers into one, like so:

```python
    # Extract and validate core [homeassistant] config
    try:
        core_config = config.pop(CONF_CORE, {})
        core_config = CORE_CONFIG_SCHEMA(core_config)
        result[CONF_CORE] = core_config

        # Merge packages
        await merge_packages_config(
            hass, config, core_config.get(CONF_PACKAGES, {}), _pack_error
        )
    except vol.Invalid as err:
        result.add_error(
            format_schema_error(hass, err, CONF_CORE, core_config),
            CONF_CORE,
            core_config,
        )
        core_config = {}

    core_config.pop(CONF_PACKAGES, None)
```

I like the previous version slightly more as I find it easier to reason about. YMMV, of course. What's your opinion? Yes, no?

Reply:
If we can't reach that code we could raise RuntimeError add a comment. @emontnemery may have an opinion.

Reply:
Nevermind, I merged both exception handlers now as described above.